File: binding/src/libqkeysequenceedit.cpp
#include <QEvent>
#include <QFocusEvent>
#include <QKeyEvent>
#include <QKeySequence>
#include <QKeySequenceEdit>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <QTimerEvent>
#include <QWidget>
#include <qkeysequenceedit.h>
#include "libqkeysequenceedit.h"
#include "libqkeysequenceedit.hxx"

QKeySequenceEdit* QKeySequenceEdit_new(QWidget* parent) {
	 return new VirtualQKeySequenceEdit(parent);
}

QKeySequenceEdit* QKeySequenceEdit_new2() {
	 return new VirtualQKeySequenceEdit();
}

QKeySequenceEdit* QKeySequenceEdit_new3(const QKeySequence* keySequence) {
	 return new VirtualQKeySequenceEdit(*keySequence);
}

QKeySequenceEdit* QKeySequenceEdit_new4(const QKeySequence* keySequence, QWidget* parent) {
	 return new VirtualQKeySequenceEdit(*keySequence, parent);
}

libqt_string QKeySequenceEdit_Tr(const char* s) {
	QString _ret = QKeySequenceEdit::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QKeySequence* QKeySequenceEdit_KeySequence(const QKeySequenceEdit* self) {
	return new QKeySequence(self->keySequence());
}

ptrdiff_t QKeySequenceEdit_MaximumSequenceLength(const QKeySequenceEdit* self) {
	return self->maximumSequenceLength();
}

void QKeySequenceEdit_SetClearButtonEnabled(QKeySequenceEdit* self, bool enable) {
	self->setClearButtonEnabled(enable);
}

bool QKeySequenceEdit_IsClearButtonEnabled(const QKeySequenceEdit* self) {
	return self->isClearButtonEnabled();
}

void QKeySequenceEdit_SetFinishingKeyCombinations(QKeySequenceEdit* self, const libqt_list finishingKeyCombinations) {
	self->setFinishingKeyCombinations(QList<QKeyCombination>());
}

libqt_list QKeySequenceEdit_FinishingKeyCombinations(const QKeySequenceEdit* self) {
	auto _ret = self->finishingKeyCombinations();
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		auto& _elem = _ret[_i];
		_data[_i] = new std::remove_reference_t<decltype(_elem)>(_elem);
	}
	return _arr;
}

void QKeySequenceEdit_SetKeySequence(QKeySequenceEdit* self, const QKeySequence* keySequence) {
	self->setKeySequence(*keySequence);
}

void QKeySequenceEdit_Clear(QKeySequenceEdit* self) {
	self->clear();
}

void QKeySequenceEdit_SetMaximumSequenceLength(QKeySequenceEdit* self, ptrdiff_t count) {
	self->setMaximumSequenceLength(count);
}

void QKeySequenceEdit_EditingFinished(QKeySequenceEdit* self) {
	self->editingFinished();
}

void QKeySequenceEdit_Connect_EditingFinished(QKeySequenceEdit* self, intptr_t slot) {
    void (*slotFunc)(QKeySequenceEdit*) = reinterpret_cast<void (*)(QKeySequenceEdit*)>(slot);
    QKeySequenceEdit::connect(self, &QKeySequenceEdit::editingFinished, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QKeySequenceEdit_KeySequenceChanged(QKeySequenceEdit* self, const QKeySequence* keySequence) {
	self->keySequenceChanged(*keySequence);
}

void QKeySequenceEdit_Connect_KeySequenceChanged(QKeySequenceEdit* self, intptr_t slot) {
    void (*slotFunc)(QKeySequenceEdit*, const QKeySequence*) = reinterpret_cast<void (*)(QKeySequenceEdit*, const QKeySequence*)>(slot);
    QKeySequenceEdit::connect(self, &QKeySequenceEdit::keySequenceChanged, [self, slotFunc](const QKeySequence& keySequence) {
	slotFunc(self, keySequence);
    });
}

libqt_string QKeySequenceEdit_Tr2(const char* s, const char* c) {
	QString _ret = QKeySequenceEdit::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QKeySequenceEdit_Tr3(const char* s, const char* c, int n) {
	QString _ret = QKeySequenceEdit::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QKeySequenceEdit_Delete(QKeySequenceEdit* self) {
    delete self;
}

File: binding/include/libqkeysequenceedit.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qkeysequenceedit.html
pub const qkeysequenceedit = struct {

    /// New constructs a new QKeySequenceEdit object.
    pub fn New(parent: ?*anyopaque) QtC.QKeySequenceEdit {
        return qtc.QKeySequenceEdit_new(@ptrCast(parent));
    }


    /// New2 constructs a new QKeySequenceEdit object.
    pub fn New2() QtC.QKeySequenceEdit {
        return qtc.QKeySequenceEdit_new2();
    }


    /// New3 constructs a new QKeySequenceEdit object.
    pub fn New3(keySequence: ?*anyopaque) QtC.QKeySequenceEdit {
        return qtc.QKeySequenceEdit_new3(@ptrCast(keySequence));
    }


    /// New4 constructs a new QKeySequenceEdit object.
    pub fn New4(keySequence: ?*anyopaque, parent: ?*anyopaque) QtC.QKeySequenceEdit {
        return qtc.QKeySequenceEdit_new4(@ptrCast(keySequence), @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QKeySequenceEdit_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qkeysequenceedit.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn KeySequence(self: ?*anyopaque, ) QtC.QKeySequence {
        return qtc.QKeySequenceEdit_KeySequence(@ptrCast(self));
    }

    pub fn MaximumSequenceLength(self: ?*anyopaque, ) i64 {
        return qtc.QKeySequenceEdit_MaximumSequenceLength(@ptrCast(self));
    }

    pub fn SetClearButtonEnabled(self: ?*anyopaque, enable: bool) void {
        qtc.QKeySequenceEdit_SetClearButtonEnabled(@ptrCast(self), enable);
    }

    pub fn IsClearButtonEnabled(self: ?*anyopaque, ) bool {
        return qtc.QKeySequenceEdit_IsClearButtonEnabled(@ptrCast(self));
    }

    pub fn SetFinishingKeyCombinations(self: ?*anyopaque, finishingKeyCombinations: []const u8) void {
        qtc.QKeySequenceEdit_SetFinishingKeyCombinations(@ptrCast(self), @ptrCast(finishingKeyCombinations));
    }

    pub fn FinishingKeyCombinations(self: ?*anyopaque, ) []const u8 {
        return qtc.QKeySequenceEdit_FinishingKeyCombinations(@ptrCast(self));
    }

    pub fn SetKeySequence(self: ?*anyopaque, keySequence: ?*anyopaque) void {
        qtc.QKeySequenceEdit_SetKeySequence(@ptrCast(self), @ptrCast(keySequence));
    }

    pub fn Clear(self: ?*anyopaque, ) void {
        qtc.QKeySequenceEdit_Clear(@ptrCast(self));
    }

    pub fn SetMaximumSequenceLength(self: ?*anyopaque, count: i64) void {
        qtc.QKeySequenceEdit_SetMaximumSequenceLength(@ptrCast(self), count);
    }

    pub fn EditingFinished(self: ?*anyopaque, ) void {
        qtc.QKeySequenceEdit_EditingFinished(@ptrCast(self));
    }

    pub fn OnEditingFinished(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QKeySequenceEdit_Connect_EditingFinished(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn KeySequenceChanged(self: ?*anyopaque, keySequence: ?*anyopaque) void {
        qtc.QKeySequenceEdit_KeySequenceChanged(@ptrCast(self), @ptrCast(keySequence));
    }

    pub fn OnKeySequenceChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QKeySequenceEdit_Connect_KeySequenceChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Event(self: ?*anyopaque, param1: ?*anyopaque) bool {
        return qtc.QKeySequenceEdit_Event(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QKeySequenceEdit_OnEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseEvent(self: ?*anyopaque, param1: ?*anyopaque) bool {
        return qtc.QKeySequenceEdit_QBaseEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn KeyPressEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QKeySequenceEdit_KeyPressEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnKeyPressEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QKeySequenceEdit_OnKeyPressEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseKeyPressEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QKeySequenceEdit_QBaseKeyPressEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn KeyReleaseEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QKeySequenceEdit_KeyReleaseEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnKeyReleaseEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QKeySequenceEdit_OnKeyReleaseEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseKeyReleaseEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QKeySequenceEdit_QBaseKeyReleaseEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn TimerEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QKeySequenceEdit_TimerEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnTimerEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QKeySequenceEdit_OnTimerEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseTimerEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QKeySequenceEdit_QBaseTimerEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn FocusOutEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QKeySequenceEdit_FocusOutEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnFocusOutEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QKeySequenceEdit_OnFocusOutEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseFocusOutEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QKeySequenceEdit_QBaseFocusOutEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QKeySequenceEdit_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qkeysequenceedit.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QKeySequenceEdit_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qkeysequenceedit.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QKeySequenceEdit_Delete(@ptrCast(self));
    }
};
